puzzles/templates/puzzles_table.html

```diff
@@ -1,150 +1,174 @@
-                <th scope="col">Slack</th>
-                <th scope="col">Tags</th>
-                <th scope="col"></th>
-                <th scope="col">Meta</th>
-            </tr>
-        </thead>
-        <tbody>
-            {% for p, status_form, puzzle_class in rows %}
-            <tr class="{{ puzzle_class }}">
-                <td scope="row">
-                    {% get_title p %}
-                </td>
-                <td><p class="text-monospace">{{ p.answer.upper }}</p></td>
-                <td>
-                    <form method="post", action="/puzzles/update_status/{{ p.pk }}/" class="form-inline">
-                        {% csrf_token %}
-                        <div class="form-group">
-                            {{ status_form.status }}
-                        </div>
-                    </form>
-                </td>
-                <td><a href="{{ p.sheet }}" target="_blank">Link</a></td>
-                <td><a href="{{ slack_base_url }}/app_redirect?channel={{ p.channel}}" target-"_blank">Link</a></td>
-                <td>
-                    {% show_tags p %}
-                </td>
-                <td>
-                    {% if p.status != "SOLVED" %}
-                        <form action="/puzzles/guess/{{ p.pk }}/" method="post" class="form-inline">
-                        {% csrf_token %}
-                        <div class="form-group">
-                            {{ guess_form.text }}
-                        </div>
-                        <button type="submit" class="btn btn-outline-primary btn-sm">Submit</button>
-                    </form>
-                     {% endif %}
-                </td>
-                <td>
-                    <button type="button" class="meta btn btn-outline-primary btn-sm" data-toggle="modal" data-target="#meta" data-pk="{{ p.pk }}">
-                        Assign Meta
-                    </button>
-                </td>
-            </tr>
-            {% endfor %}
-        </tbody>
-        <tfoot>
-            <tr>
-                <th scope="col" class="column_searchable">Name</th>
-                <th scope="col" class="column_searchable">Answer</th>
-                <th scope="col" class="column_searchable">Status</th>
-                <th scope="col"></th>
-                <th scope="col"></th>
-                <th scope="col" class="column_searchable">Tags</th>
-                <th scope="col"></th>
-                <th scope="col"></th>
-            </tr>
-        </tfoot>
-    </table>
-</div>
+<table class="table table-sm tree" id="puzzles" style="width:100%">
+</table>
 
 <script type="text/javascript">
+var slack_base_url = '{{ slack_base_url }}';
+var data = [
+{% for p, tags, metas, active_users, status_form, puzzle_class in rows %}['{{p.pk}}', '{{p.name}}', '{{p.url}}', '{{p.is_meta}}', '{{active_users|join:", "}}', '{{p.answer}}', '{{p.status}}', '{{p.sheet}}', '{{p.channel}}', {{tags|safe}}, '', '{{puzzle_class}}', ],
```

Comment:
No need to change for now, but in a future change, it'd make sense to expose an endpoint to just fetch the JSON data (rather than the page and having the data be delivered as part of the .html in a script tag)

Reply:
Yeah, I was thinking the same thing. Shouldn't be too hard to add such an endpoint, and then we should be able to refresh the table without an entire page reload. And we can set a pollInterval of, say, 5-10 seconds so the table refreshes automatically.